### ISSUE TITLE
Config comments sometimes ignore incorrect options

### ISSUE THREAD

**Issue by btmills** [eslint/eslint#2725]
Credit goes to @gyandeeps for catching this one:

```js
/* eslint vars-on-top:"aaalways", semi:true */
function foo() {
    foo()
    var bar = 42
}
```

```sh
$ eslint --reset --no-eslintrc test.js
# No errors reported
```

[`eslint.js` line 290](https://github.com/eslint/eslint/blob/d007063bddc3674cac6c09ace68fb692d36d3116/lib/eslint.js#L290) was causing the incorrect configuration to be ignored because `semi:true` failed the filter.

This will be fixed by removing that line entirely and letting the validator catch any errors.

Working on this.
